services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java

```diff
@@ -470,6 +470,33 @@ public void addBearing_doesGetFormattedInUrlCorrectly() throws Exception {
       directions.cloneCall().request().url().queryParameter("bearings"));
   }
 
+  @Test
+  public void addNullBearings_doesGetFormattedInUrlCorrectly() throws Exception {
+    List<Double> bearing1 = new ArrayList<>();
+    bearing1.add(45d);
+    bearing1.add(90d);
+
+    List<Double> bearing2 = new ArrayList<>();
+    bearing2.add(2d);
+    bearing2.add(90d);
+
+    List<List<Double>> bearings = new ArrayList<>();
+    bearings.add(null);
+    bearings.add(null);
+    bearings.add(bearing1);
+    bearings.add(bearing2);
+    bearings.add(null);
+
+    MapboxDirections directions = MapboxDirections.builder()
+        .destination(Point.fromLngLat(13.4930, 9.958))
+        .origin(Point.fromLngLat(1.234, 2.345))
+        .bearings(bearings)
+        .accessToken(ACCESS_TOKEN)
+        .build();
+    assertEquals(";;45,90;2,90;",
```

Comment:
Debugging this PR I found that `ParseUtils#parseToListOfListOfDoubles` in this concrete example is not working as expected

```java
@Test
public void checksParseToListOfListOfDoublesEmptyTrailing() {
  List<List<Double>> bearings = ParseUtils.parseToListOfListOfDoubles(";;45,90;2,90;");

  assertEquals(5, bearings.size());
}
```

It should return a list of 5 elements and it's returning a list of 4 because of not splitting with trailing empty strings https://github.com/mapbox/mapbox-java/blob/123e7cd6c7aac9b125ed2a679c09f68b6e5dadd7/services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/ParseUtils.java#L141

After changing to `original.split(SEMICOLON, -1);` other tests start failing. We should revisit the empty trailing scenarios for other fields too.

Reply:
@Guardiola31337 @LukasPaczos 
when you parse a `";;45,90;2,90;"` with `ParseUtils#parseToListOfListOfDoubles` you get a list with size 4 because it removes trailing empty strings. And it worked the same way  before refactoring, `split(SEMICOLON)` is used in both cases

Reply:
so, it works as expected

Reply:
tested it with @kmadsen , @Guardiola31337 you are right, `bearing` string should HAVE trailing `;` if we pass null bearings. 
I'll fix logic and tests

Reply:
@korshaknn is `";;45,90;2,90;"` and `";;45,90;2,90"` going to be equal in this case? I think it shouldn't be. We should take the strategy that Directions API uses:
- what happens if we pass `"45,90;"` as the request parameter? Is it considered as 2 values (1 empty)?
- what happens if we pass `"45,90;;"` as the request parameter? Is it considered as 3 values (2 empty)?
If yes, we shouldn't trim the trailing empty string.

/cc @danpaz

Reply:
Oh, looks like we commented at the same time @korshaknn :) Thanks for verifying!

Reply:
@LukasPaczos 
`";;45,90;2,90;"` will be valid for 5 coordinates
`";;45,90;2,90"` will be valid for 4 coordinates